OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -264,18 +262,45 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 			var barrel = Weapon.Burst == 1 ? Barrels[currentBarrel] : Barrels[Burst % Barrels.Length];
 			currentBarrel++;
 
-			FireBarrel(self, facing, target, barrel);
-
+			FireBarrel(self, target, barrel);
 			UpdateBurst(self, target);
 
-			return barrel;
+			return true;
 		}
 
-		protected virtual void FireBarrel(Actor self, IFacing facing, Target target, Barrel barrel)
+		protected virtual void FireBarrel(Actor self, Target target, Barrel barrel)
 		{
 			foreach (var na in notifyAttacks)
 				na.PreparingAttack(self, target, this, barrel);
 
+			ScheduleDelayedAction(Info.FireDelay, () =>
+			{
+				var args = CreateProjectileArgs(self, target, barrel);
+				if (args == null)
+					return;
+
+				var projectile = args.Weapon.Projectile != null ? args.Weapon.Projectile.Create(args) : null;
```

Comment:
This can now simplify to `var projectile = args.Weapon.Projectile?.Create(args);`